src/test/java/net/consensys/orion/http/handler/SendRawHandlerTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.orion.http.handler;
+
+import static net.consensys.cava.crypto.Hash.sha2_512_256;
+import static net.consensys.cava.io.Base64.encodeBytes;
+import static net.consensys.orion.http.server.HttpContentType.APPLICATION_OCTET_STREAM;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import net.consensys.cava.crypto.sodium.Box;
+import net.consensys.cava.junit.TempDirectory;
+import net.consensys.orion.enclave.EncryptedPayload;
+import net.consensys.orion.enclave.sodium.MemoryKeyStore;
+import net.consensys.orion.helpers.FakePeer;
+import net.consensys.orion.http.server.HttpContentType;
+
+import java.nio.file.Path;
+import java.security.Security;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.mockwebserver.MockResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SendRawHandlerTest extends HandlerTest {
+
+  protected MemoryKeyStore memoryKeyStore;
+
+  static {
+    Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+  }
+
+  @BeforeEach
+  void setUpKeyStore(@TempDirectory final Path tempDir) {
+    memoryKeyStore = new MemoryKeyStore();
+  }
+
+  @Test
+  void sendingWithARawBody() throws Exception {
```

Comment:
could this test be way simpler with a mocked distributePayloadManager - and confirm it gets invoked with the right data - then that the mocked response is OCTET encoded?

Reply:
This test is still useful - but more of an integrationTest

Reply:
It could. What I did was just move the logic related to /sendraw that was in the SendHandlerTest to this SendHandlerRawTest. I thought that the effort of mocking + moving the tests to integration tests atm wouldn't pay off.

Can we leave it this way for now?